### PR TITLE
Add sentry integration

### PR DESCRIPTION
**Pull request by misaka**
Generic Sentry integration added. Relies on `SENTRY_DSN` being set in the deployed environment.

This change has been tested, however we don't have a team account on Sentry. At the moment I'm just using a DSN for the organisation "Good Machine" setup on my account; to invite other members to it will require a team, business or open source account, so far as I can see.

### CODE REVIEW

**Comment by misaka** [comment]
I've added a `SENTRY_DSN` for the Heroku staging environment so in theory this should work once deployed. It might be worth adding to the review apps to check it's functionality there.

File: config/initializers/sentry.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+Sentry.init do |config|
+  config.breadcrumbs_logger = [:active_support_logger, :http_logger]
+
+  config.traces_sample_rate = 1.0
+
+  filter = ActiveSupport::ParameterFilter.new(Rails.application.config.filter_parameters)
+  config.before_send = lambda do |event, _hint|
+    filter.filter(event.to_hash)
+  end
```
Comment (tvararu) [inline config/initializers/sentry.rb:11]: I recommend that we prevent Sentry from logging for 422 and "user errors," which can be done with an excluded exceptions array like this:

https://github.com/DFE-Digital/find-a-lost-trn/blob/main/config/initializers/sentry.rb#L26-L34

Comment (misaka) [reply config/initializers/sentry.rb:11]: Would it maybe be worth keeping these while we're in the model office & pilot phase? These could be interesting, and it's hidden behind auth so we shouldn't get spurious script kiddies and vulnerability scanning.

Comment (tvararu) [reply config/initializers/sentry.rb:11]: Hmm, yeah, we can revisit this later :+1:

**Review by tvararu — APPROVED** [review]

**Comment by benilovj** [comment]
Do we need to blank out PII fields? I think we should...

**Comment by misaka** [comment]
Line 9-11 is setting up the filtering out of parameters. This depends on [/config/initializers/filter_parameter_logging.rb](https://github.com/nhsuk/record-childrens-vaccinations/blob/main/config/initializers/filter_parameter_logging.rb) being setup.

We can add that to this PR.

**Pull request merged**